tipg/middleware.py

```diff
@@ -51,19 +51,38 @@ async def dispatch(self, request: Request, call_next):
 class CatalogUpdateMiddleware(BaseHTTPMiddleware):
```

Comment:
I thought the biggest thing you wanted to do was to get rid of BaseHTTPMiddleware???

Reply:
I will in an other PR